Traits/Reflection.php

```diff
@@ -81,4 +82,28 @@ public function getMethods() {
 	public function getProperties() {
 		return $this->_getProperties();
 	}
+
+	/**
+	 * (non-PHPdoc)
+	 * @see \Circle\UtilityBundle\Traits\Reflection::_getTraits()
+	 */
+	public function getTraits($deep = false) {
```

Comment:
Im Moment brauche ich einfach alle Traits. Wenn ich sage @Uses(Trait), dann will ich wissen, ob die Klasse iiiirgendwo den Trait geused hat. Ob das jetzt direkt oder indirekt ist ist mir scheiß egal.
Da allerdings der ReflectionTrait exakt die Methoden abbildet, die man von der ReflectionClass kennt, müsste getTraits() nur die Traits der Klasse und nicht die des Parent usw zurückgeben. Daher die Idee, ne weitere Funktion einzuführen, die dann iwie anders heißt. Nur getAllTraits oder was weiß ich finde ich alles Müll. Daher die Idee, hier über nen Parameter zu kontrollieren, was der tut. Deep ja/nein fand ich am Treffendsten.

Problem bei dem Baum + Ebene: Ich habe mehrere Stellen, an denen ich grabe:
- Inheritance
- uses in traits

z. B. class->getTraits->getTraits->getTraits + class->getParent()->getTraits()->getTraits()

letztendlich ist das schon ein Baum, nur da genau zu sagen was welche Ebene ist ist nicht nachvollziehbar und daher nicht wirklich sinnvoll als API, weil kein Mensch rafft, was Ebene 1 usw bedeutet. Bei dem Beispiel oben wäre ja schon die Frage ob die jetzt auf der selben Ebene sind oder nicht. Im Grunde schon, aber in einem anderen Zweig.